lxd/instance/drivers/driver_qemu.go

```diff
@@ -7427,21 +7443,75 @@ func (d *qemu) DeviceEventHandler(runConf *deviceConfig.RunConfig) error {
 }
 
 // vsockID returns the vsock Context ID for the VM.
-func (d *qemu) vsockID() int {
-	// We use the system's own VsockID as the base.
-	//
-	// This is either "2" for a physical system or the VM's own id if
-	// running inside of a VM.
-	//
-	// To this we add 1 for backward compatibility with prior logic
-	// which would start at id 3 rather than id 2. Removing that offset
-	// would cause conflicts between existing VMs until they're all rebooted.
-	//
-	// We then add the VM's own instance id (1 or higher) to give us a
-	// unique, non-clashing context ID for our guest.
+func (d *qemu) vsockID() (int, error) {
+	// Return an already existing Context ID.
+	existingVsockID, ok := d.localConfig["volatile.vsock_id"]
+	if ok {
+		vsockID, err := strconv.Atoi(existingVsockID)
+		if err != nil {
+			return 0, fmt.Errorf("Failed to convert Context ID: %w", err)
+		}
 
-	info := DriverStatuses()[instancetype.VM].Info
-	return info.Features["vhost_vsock"].(int) + 1 + d.id
+		return vsockID, nil
+	}
+
+	instanceUuid := uuid.Parse(d.localConfig["volatile.uuid"])
+	if instanceUuid == nil {
+		return 0, fmt.Errorf("Instance does not have a UUID")
+	}
+
+	// Pick the first 4 bytes of the random UUID (v4) to create an uint32 representation.
+	// Since the instances UUID already exist, we don't need to create another random value.
+	firstBytes := binary.LittleEndian.Uint32(instanceUuid[:4])
+
+	var i uint32
```

Comment:
If we don't explicitly set it to `uint32`, in the loop every `i` needs to be casted since `firstBytes` is already `uint32`. For the logic of the loop it is required that `candidateVsockID` can overflow after the max value for `uint32` is exceeded. So we can't change the type to just `int`.

Reply:
I see. But this should work

```go
for i := uint32(0); ...
```

Reply:
Changed.